common/src/main/java/us/ilite/common/LoggedData.java

```diff
@@ -0,0 +1,148 @@
+
+    public LoggedData() {
+        //Add new codexes as we support more into this list
+        mLoggedCodexes = Arrays.asList(
+            new CodexNetworkTablesParser<EGyro>(imu),
+            new CodexNetworkTablesParser<EDriveData>(drive),
+            new CodexNetworkTablesParser<ELogitech310>(driverinput, "DRIVER"),
+            new CodexNetworkTablesParser<ELogitech310>(operatorinput, "OPERATOR")
+        );
+
+        //This loop makes a Writer for each parser and sticks it into mWriters
+        for (CodexNetworkTablesParser parser : mLoggedCodexes) {
+            try {
+                File file = parser.file();
+                handleCreation(file);
+                mWriters.put(parser.getCSVIdentifier(), new BufferedWriter(new FileWriter(parser.file())));
+            }
+            catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * Translate NT to on-computer codex for each CodexNetworkTablesParser in mLoggedCodexes
+     */
+    public void logFromNetworkTables() {
+        mLoggedCodexes.forEach(c -> c.parseFromNetworkTables());
+    }
+
+    /**
+     * Makes a csv file and writes the starting row/header for each CodexNetworkTablesParser in mLoggedCodexes
+     */
+    public void logFromCodexToCSVHeader() {
+        for (CodexNetworkTablesParser parser : mLoggedCodexes) {
+            try {
+                Writer logger = mWriters.get(parser.getCSVIdentifier());
+                logger.append(parser.codexToCSVHeader());
+                logger.flush();
+            }
+            catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+    /**
+     * Logs codex values to its corresponding csv
+     */
+    public void logFromCodexToCSVLog() {
+        for (CodexNetworkTablesParser parser : mLoggedCodexes) {
+            try {
+                Writer logger = mWriters.get(parser.getCSVIdentifier());
+                logger.append(parser.codexToCSVLog());
+                logger.flush();
+            }
+            catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+    /**
+     * Closes all the writers in mWriters
+     */
+    public void closeWriters() {
+        for (Writer writer : mWriters.values()) {
+            try {
+                writer.close();
+            }
+            catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * Makes the log file if it doesn't already exist
+     */
+    private void handleCreation(File pFile) throws IOException {
+        //Makes every folder before the file if the CSV's parent folder doesn't exist
+        if(!pFile.getParentFile().exists()) pFile.getParentFile().mkdirs();
+
+        //Creates the .CSV if it doesn't exist
+        if(!pFile.exists()) pFile.createNewFile();
+    }
+
+    //For testing purposes
+    public static void main(String[] args) {
```

Comment:
What if I keep the *Data* class and just rename it to *RobotData* (and keep *LoggedData*)? This is because *Data* is needed in *robot\* to send the robot's codexes to NTs and has some fields that are referenced in other robot classes.